e2e_tests/tests/all_providers/multitenancy.rs

```diff
@@ -66,4 +85,18 @@ fn client1_after() {
         client.set_provider(*provider);
         client.destroy_key(key.clone()).unwrap();
     }
+
+    client
+        .generate_rsa_sign_key("client1-key".to_string())
+        .unwrap();
+    let mut clients = client.list_clients().unwrap();
+    assert_eq!(clients.len(), 2);
+    client.delete_client(clients.remove(0)).unwrap();
+    let mut clients = client.list_clients().unwrap();
+    assert_eq!(clients.len(), 1);
+    client.delete_client(clients.remove(0)).unwrap();
+    let clients = client.list_clients().unwrap();
+    assert_eq!(clients.len(), 0);
+    let keys = client.list_keys().unwrap();
+    assert_eq!(keys.len(), 0);
```

Comment:
It can not directly be done here for all authentication methods, it would involve adding one more function `client2_after` to be executed under the other clients and one more line in the `ci.sh` file. Can do it if you think it's worth it?

Reply:
Nah, leave it as is, I think there are other tests for stuff like this, right?